tox-offline-draft.md

```diff
@@ -5,77 +5,146 @@
 This proposal aims to provide a general and technical walkthrough of an
 offline messaging protocol for use with the Tox protocol using its existing
 Distributed Hash Table (DHT). Messages are only stored in the DHT when the
-message receiver is currently offline. The sender does not have to be
-online for the receiver to retrieve messages.
+message receiver is currently offline. The sender does not have to be online
+for the receiver to retrieve messages.
 
 As this method uses the DHT, it is currently UDP only and will not work
 unless UDP is enabled. This restriction will hopefully be lifted with
 further work and development of the Tox protocol.
 
+## Terminology
+PoW:
+  Proof-of-Work -- Determines CPU work necessary in order to store an offline message on the DHT.
+
+SENDER:
+  The friend of RECEIVER sending an offline message to her when she is offline.
+
+RECEIVER:
+  The friend of SENDER receiving an offline message.
+
+STORAGE_NODE:
+  One of potentially many nodes storing offline messages for other users.
+
+OM Keys:
+  Offline Messaging public / private keypair.
+
 ## DHT Extension
 
-The DHT must be extended in order to be able to store arbitrary data. Tox
-Clients should have a reasonable default for DHT storage size set and
+The DHT must be extended in order to be able to store offline messages. Tox
+Clients should have a reasonable default set for DHT storage size and
 optionally allow users to change it. This space will be used for storing
 messages of other users.
```

Comment:
In my earlier notes I had nodes clear offline messages on boot up as they may have expired already anyways, however if the TTL isn't met we might as well keep hosting them.

Another issue we need to design for is telling all storage_nodes that a message has been requested already even if they were offline when the request happened.